pkgs/development/libraries/libnbd/default.nix

```
@@ -4,6 +4,7 @@
 , bash-completion
 , pkg-config
 , perl
+, withPython ? true, python3
```

```suggestion
, withPython ? true
, python3
```

---

Can this be unconditional?

---

Sorry, I don't quite understand the question? Python is only pulled in as a `buildInput` if `withPython` is set to `true` (to allow a user to override it if necessary for some reason). Is there other desired behaviour?

---

In a certain sense, Python is employed here to create the bindings.
But the bindings would be created anyway - Hydra cache will contain both nbd and the python-nbd.
(IIRC the toPythonModule works mostly like a thin wrapper.)

So I don't see why not to build this bindings unconditionally.

---

Hmm, I was thinking more along the lines of a user on a nonstandard `buildPlatform` compiling it themselves. For the purposes of Hydra it does indeed make no difference since we default the setting to `true` and don't override it to `false` anywhere. I feel like it's just a general good practice to provide the option to not pull in an entire language ecosystem as a build dependency since upstream made the bindings optional, although I recognise that this is less applicable with python being so ubiquitous that it's probably getting pulled in somewhere else anyway. If you still want it, I can go ahead and make it unconditional.

---

That being said, a better name for this variable would be `buildPythonModule` instead of `withPython`.
```suggestion
, buildPythonModule ? true
, python3
```

---

How about just `pythonBindings`?

---

This is a small hint from an [old book](https://www.amazon.com/Practice-Programming-Addison-Wesley-Professional-Computing/dp/020161586X) about naming ~~variables~~ attributes:
Boolean variables should have Boolean-ish names.

In Scheme this is easy: just add a question mark at the end.
But this is not Guix, so we need another convention.

The purpose of this attribute is to control if the Python module will be built. Then a good name would be `buildPythonModule`, or maybe `buildPythonBindings` (to not clash with the already used function name `buildPythonModule`). 

Think on this like asking a question, "do you want to build the Python bindings?" Yes = true, No = false.

On the other hand, the other two are a bit ambiguous:

- `withPython` looks like it will bring all Python ecosystem bundled
- `pythonBindings` looks like a list of Python bindings provided after the build.